server/implementation/src/main/java/io/smallrye/graphql/bootstrap/Bootstrap.java

```diff
@@ -214,9 +219,20 @@ private void generateGraphQLSchema() {
 
         if (Config.get().isFederationEnabled()) {
             log.enableFederation();
+
+            if (!(schema.hasQueries() || schema.hasResolvers() || schema.hasMutations() || schema.hasSubscriptions())) {
+                throw new RuntimeException("You must define at least one method marked with one of the annotations - "
+                        + "@Query, @Mutation, @Subscription, @Resolver. An empty diagram has no meaning");
+            }
+
+            // hack! For schema build success if queries are empty.
+            // It will be overrides in Federation transformation
+            addSdlQueryToSchema(schemaBuilder, queryRootType);
```

Comment:
I‘m not sure, if I got this right: graphql-java federation transformation is going to overwrite the `_service` query that we add here? It‘s just a dummy that is required to make the schema builder happy? Maybe it would make the code flow more understandable, if the method was named `addDummyQuery`. What do you think?

Reply:
yes, you right, i rename this